packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -3016,3 +3155,266 @@ typedef SelectableRegionContextMenuBuilder = Widget Function(
+  }) : _selectionNotifier = selectionNotifier {
+    assert(
+      !_selectionNotifier.registered,
+      'The SelectionListenerNotifier provided is already attached to another SelectionListener. Try providing a new SelectionListenerNotifier.',
+    );
+    _selectionNotifier._selectionDelegate = this;
+  }
+
+  SelectionListenerNotifier _selectionNotifier;
+  void setNotifier(SelectionListenerNotifier newNotifier) {
+    _selectionNotifier = newNotifier;
+    _selectionNotifier._selectionDelegate = this;
+  }
+
+  SelectionGeometry? _previousSelectionGeometry;
+
+  @override
+  SelectionResult dispatchSelectionEvent(SelectionEvent event) {
+    final SelectionResult result = super.dispatchSelectionEvent(event);
+    if (_previousSelectionGeometry != value) {
+      _selectionNotifier.notifyListeners();
+      _previousSelectionGeometry = value;
+    }
+    return result;
+  }
+
+  @override
+  void dispose() {
+    _selectionNotifier._unregisterSelectionListenerDelegate();
+    super.dispose();
+  }
+
+  @override
+  SelectedContentRange? get range => getSelection();
+
+  @override
+  SelectionStatus get status => value.status;
+}
+
+/// A read-only interface for accessing the details of a selection under a [SelectionListener].
+///
+/// This includes information such as the status of the selection indicating
+/// if it is collapsed or uncollapsed, the [SelectedContentRange] that includes
+/// the start and end offsets of the selection local to the [SelectionListener]
+/// that reports this object.
+///
+/// This object is typically accessed by providing a [SelectionListenerNotifier]
+/// to a [SelectionListener] and retrieving the value from [SelectionListenerNotifier.selection].
+abstract final class SelectionDetails {
+  /// The computed selection range of the owning [SelectionListener]s subtree.
+  ///
+  /// Returns `null` if there is nothing selected.
+  SelectedContentRange? get range;
+
+  /// The status that indicates whether there is a selection and whether the selection is collapsed.
+  SelectionStatus get status;
+}
+
+/// Notifies listeners when the selection under a [SelectionListener] has been
+/// changed.
+///
+/// This object is typically provided to a [SelectionListener].
+final class SelectionListenerNotifier extends ChangeNotifier {
+  _SelectionListenerDelegate? _selectionDelegate;
+
+  /// The details of the selection under the [SelectionListener] that owns this notifier.
+  ///
+  /// Throws an exception if this notifier has not been registered to a [SelectionListener].
+  /// To check if a notifier has been registered to a [SelectionListener] use [registered].
+  SelectionDetails get selection => _selectionDelegate ?? (throw Exception('Selection client has not been registered to this notifier.'));
+
+  /// Whether this [SelectionListenerNotifier] has been registered to a [SelectionListener].
+  bool get registered => _selectionDelegate != null;
+
+  void _unregisterSelectionListenerDelegate() {
```

Comment:
it seems weird that we have this api but no `registerSelectionListenerDelegate()` I would create a matching API instead of assigning the _selectionDelegate directly in the _SelectionListenerDelegate

Reply:
Will update with matching register API.